File: 2015/day1.py
import AOCInit
import util

inp = util.getInput(d=1, y=2015)
seq = util.sub('()', (1, -1), inp)

# part 1
print(sum(seq))

# part 2
print(util.firstAccumSuchThat(seq, lambda x, y: x + y, lambda x: x < 0)[0] + 1)

File: 2015/day1.rb
inp = File.read('input1').chars.map { |x| { '(' => 1, ')' => -1 }[x] }

# part 1
puts inp.sum

# part 2
(1...inp.size).to_a.each_with_index { |e, i| inp[e] += inp[i] }
puts (0...inp.size).find { |i| inp[i] == -1 } + 1
